prov/gni/src/gnix_vc.c

```diff
@@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
 	      sizeof(gni_mem_handle_t);
 	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
 
-	memcpy(cptr, &rtype, sizeof(rtype));
+	*((uint8_t *)cptr) = rtype;
 	cptr += sizeof(rtype);
```

Comment:
This might not be relevant to this commit, but why add the possibility of typo errors? We should make this into